cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/plugin/TrackingMacroEvaluator.java

```diff
@@ -43,6 +59,22 @@ public String evaluate(String macroFunction, String... arguments) {
   }
 
   /**
+   * If the last string evaluated by the evaluator's associated {@link MacroParser} contained a macro,
+   * returns the default value of the property's type. Otherwise, returns the original property string.
+   * @param originalPropertyString the string last evaluated
+   * @param propertyName the name of the property last evaluated
+   * @param propertyType the type of the property last evaluated
+   * @return
+   */
+  public String getValue(String originalPropertyString, String propertyName, String propertyType) {
```

Comment:
I was talking with @gsps1 and the intention was to reduce code reuse in PluginInstantiator. That makes sense though.. we can still remove the code reuse even if it's in PluginInstantiator. I'll move it back.